Lib/random.py

```diff
@@ -221,22 +242,23 @@ def randint(self, a, b):
 
         return self.randrange(a, b+1)
 
-    def _randbelow(self, n, int=int, maxsize=1<<BPF, type=type,
-                   Method=_MethodType, BuiltinMethod=_BuiltinMethodType):
+    def _randbelow_with_getrandbits(self, n):
```

Comment:
Why not just call this _randbelow and only patch the without getrandbits case?

Reply:
You need to keep a reference to the "with getrandbits" implementation to be able to return to using it in the case:

```
class Rand1(Random):
    def random(self): ...
        # _randbelow should use random()

class Rand2(Rand1):
    def getrandbits(self): ...
         # _randbelow should use getrandbits() now again
```

Reply:
Okay, I see what you're trying to do.